template/istqb.cls

```diff
@@ -106,44 +107,23 @@
         \g_istqb_schema_tl
     }
     { \\ \leavevmode \kern 0.1in }
+  % Level
   \tl_if_empty:NF
     \g_istqb_level_tl
-    {
-      \mbox { \g_istqb_level_tl }
-      \tl_if_empty:NF
-        \g_istqb_type_tl
-        { ~ }
-    }
-  \tl_if_empty:NF
-    \g_istqb_type_tl
-    { \mbox { \g_istqb_type_tl } }
-  \bool_if:nF
-    {
-      (
-        \tl_if_empty_p:N
-          \g_istqb_level_tl &&
-        \tl_if_empty_p:N
-          \g_istqb_type_tl
-      ) ||
-      (
-        \tl_if_empty_p:N
-          \g_istqb_title_tl &&
-        \tl_if_empty_p:N
-          \g_istqb_code_tl
-      )
-    }
-    { \,--\, }
+    { \mbox { \g_istqb_level_tl } }
+  % Title (code)
   \tl_if_empty:NF
     \g_istqb_title_tl
     {
-      \mbox { \g_istqb_title_tl }
+      ~\mbox { \g_istqb_title_tl }
       \tl_if_empty:NF
         \g_istqb_code_tl
-        { ~ }
+        { ~\mbox { ( \g_istqb_code_tl ) } }
     }
+  % Type
   \tl_if_empty:NF
-    \g_istqb_code_tl
-    { \mbox { ( \g_istqb_code_tl ) } }
+    \g_istqb_type_tl
+    { ~\mbox { \g_istqb_type_tl } }
```

Comment:
If the level is empty/undefined, there will be an extra space before the title/type due to the `~` before `\mbox` on lines 118 and 126.

If the level can't be empty/undefined, we should a) make it mandatory in the YAML schema `schema/metadata.yml` and b) remove the needless `\tl_if_empty:NF \g_istqb_level_tl` on lines 111-113 above.

If the level can be empty/undefined, there should be c) `\tl_if_empty:NF \g_istqb_level_tl { ~ }` immediately before the printing of the title on line 118 and d) `\bool_if:nF { \tl_if_empty_p:N \g_istqb_level_tl && \tl_if_empty_p:N \g_istqb_title_tl } { ~ }` immediately before the printing of the type on line 126.

@danopolan: Whichever solution is correct?